### PR TITLE
Ng all events

### PR DESCRIPTION
**Pull request by Nikolay-Gavazov**
Add cancel button styles.Create get current user function.Create profile page, and made routing work.

### CODE REVIEW

File: src/app/user/profile/profile.page.html [hunk 1]
```diff
@@ -1,14 +1,68 @@
...
-      <ion-title size="large">Profile</ion-title>
-    </ion-toolbar>
-  </ion-header>
-  <app-explore-container name="Profile"></app-explore-container>
+  <app-header />
+  <div collapse="condense">
+    <div class="heading-title">
+      <h1>Профил</h1>
+      <p>
+        {{user.role === "regular" ? 'на потребител' : "на организатор"}}
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html]: What if an admin?

File: src/app/user/profile/profile.page.html [hunk 2]
```diff
@@ -1,14 +1,68 @@
...
+      </p>
+    </div>
+  </div>
+  <div class="container">
+    <form #userEditForm="ngForm" (submit)="onUserEdit(userEditForm)">
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html:14]: This is only valid if the user is regular. What if an organizer ?

File: src/app/user/profile/profile.page.ts [hunk 3]
```diff
@@ -1,12 +1,48 @@
...
+import { AuthService } from 'src/shared/services/auth.service';
 
 @Component({
   selector: 'app-profile',
   templateUrl: './profile.page.html',
   styleUrls: ['./profile.page.scss'],
 })
 export class ProfilePage implements OnInit {
-  constructor() {}
+  user: any = {email: '', firstName: '', lastName: '', region: ''};
```
Comment (Getsov) [inline src/app/user/profile/profile.page.ts]: The user must be of type User

File: src/app/user/profile/profile.page.html [hunk 4]
```diff
@@ -1,14 +1,68 @@
...
+      <div class="auth-wrapper">
+        <div class="auth-input">
+          <ion-input type="email" label="Имейл*" labelPlacement="floating" name="email" ngModel [ngModel]="user.email"
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html]: Add TODO for the email change overlay

File: src/app/user/profile/profile.page.html [hunk 5]
```diff
@@ -1,14 +1,68 @@
...
+            #email="ngModel" clearInput required
+            pattern="^[a-zA-Z0-9\._+\-]+@[a-zA-Z0-9\.]+\.[a-zA-Z]{2,}$"></ion-input>
+        </div>
+
+        <div class="auth-input">
+          <ion-input type="password" label="Парола*" labelPlacement="floating" name="password" ngModel
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html:24]: Add TODO for the password change overlay

File: src/app/user/profile/profile.page.html [hunk 6]
```diff
@@ -1,14 +1,68 @@
...
+            #password="ngModel" [ngModel]="password" required minlength="6" maxlength="24" clearInput
+            pattern="^\S+$"></ion-input>
+        </div>
+
+
+        <div class="auth-input">
+          <ion-input type="text" label="Име" labelPlacement="floating" clearInput name="firstName" ngModel
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html]: What if an organizer?

File: src/app/user/profile/profile.page.html [hunk 7]
```diff
@@ -1,14 +1,68 @@
...
+            #firstName="ngModel" [ngModel]="user.firstName" pattern="^[a-zA-Zа-яА-Я \-]+$"></ion-input>
+        </div>
+
+        <ng-container *ngIf="firstName.touched">
+          <p class="auth-validation-error-message" *ngIf="firstName.errors?.['pattern']">
+            Неползволен символ в името!
+          </p>
+        </ng-container>
+
+        <div class="auth-input">
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html]: What if an organizer?

File: src/app/user/profile/profile.page.html [hunk 8]
```diff
@@ -1,14 +1,68 @@
...
+          <ion-select interface="popover" placeholder="Населено място" class="region-select" slot="end" name="region"
+            #region (ionChange)="onRegionChange($event.detail.value)" [ngModel]="user.region">
+            <ion-select-option value="{{region}}" *ngFor="let region of bulgarianRegions">{{region}}</ion-select-option>
+          </ion-select>
+        </div>
+
+      </div>
+
+      <div class="main-button-wrapper">
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html:94]: Add TODO for functionalities

File: src/app/user/profile/profile.page.html [hunk 9]
```diff
@@ -1,14 +1,68 @@
...
+        <button class="main-button">Редактирай профила</button>
+      </div>
+      <div class="cancel-button-wrapper">
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html:98]: Add TODO for functionalities

File: src/app/user/profile/profile.page.ts [hunk 10]
```diff
@@ -1,12 +1,48 @@
...
+  bulgarianRegions: string[] = Object.keys(BulgarianRegions).filter((v) =>
+    isNaN(Number(v))
+  );
+  editResponseError!: string;
+  region!: string;
```
Comment (Getsov) [inline src/app/user/profile/profile.page.ts]: why with "!" ?

File: src/app/user/profile/profile.page.ts [hunk 11]
```diff
@@ -1,12 +1,48 @@
...
+
+    const { email, password, firstName, lastName } = userEditForm.value;
+
+    const region = this.region;
+  }
+  constructor(private authService: AuthService) {}
+
+  ngOnInit() {
+    if(this.authService.getUser()){
+      this.user = this.authService.getUser();
```
Comment (Getsov) [inline src/app/user/profile/profile.page.ts]: If there is none, get none? What is this logic about?

File: src/app/user/profile/profile.page.html (hunk 1 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html]: In figma there is only a user and an organizer.

File: src/app/user/profile/profile.page.ts (hunk 3 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.ts]: done

File: src/app/user/profile/profile.page.ts (hunk 10 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.ts]: its same like region: string = "";

File: src/app/user/profile/profile.page.ts (hunk 11 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.ts]: Redirect to login?

File: src/app/user/profile/profile.page.html (hunk 9 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html:98]: done

File: src/app/user/profile/profile.page.html (hunk 8 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html:94]: done

File: src/app/user/profile/profile.page.html (hunk 7 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html]: done

File: src/app/user/profile/profile.page.html (hunk 6 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html]: done

File: src/app/user/profile/profile.page.html (hunk 5 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html:24]: done

File: src/app/user/profile/profile.page.html (hunk 4 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html]: done

File: src/app/user/profile/profile.page.html (hunk 2 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html:14]: I made the form inside with a check if it is a user or organizer.

File: src/app/user/profile/profile.page.ts [hunk 12]
```diff
@@ -10,12 +11,18 @@ import { AuthService } from 'src/shared/services/auth.service';
   styleUrls: ['./profile.page.scss'],
 })
 export class ProfilePage implements OnInit {
-  user: any = {email: '', firstName: '', lastName: '', region: ''};
+  user: User | null = {
+    email: '', firstName: '', lastName: '', region: '',
+    role: '',
+    organizatorName: '',
```
Comment (Getsov) [inline src/app/user/profile/profile.page.ts:17]: If it isn’t an organizer, there shouldn’t be any organizatorName. You don’t have to pass an empty Uset object with all properties, if it’s the main goal here

File: src/app/user/profile/profile.page.html [hunk 13]
```diff
@@ -4,51 +4,85 @@
     <div class="heading-title">
       <h1>Профил</h1>
       <p>
-        {{user.role === "regular" ? 'на потребител' : "на организатор"}}
+        {{user?.role === "regular" ? 'на потребител' : "на организатор"}}
```
Comment (Getsov) [inline src/app/user/profile/profile.page.html]: Add the admin and be consistent with the quotes - “ ‘ “ and “ “ “

File: src/app/user/profile/profile.page.ts [hunk 14]
```diff
@@ -40,9 +47,9 @@ export class ProfilePage implements OnInit {
       this.email = this.user["email"];
       this.firstName = this.user.firstName;
       this.lastName = this.user.lastName; */
+    }else{
```
Comment (Getsov) [inline src/app/user/profile/profile.page.ts]: Spaces around

File: src/app/user/profile/profile.page.ts [hunk 15]
```diff
@@ -27,11 +34,11 @@ export class ProfilePage implements OnInit {
...
-    const { email, password, firstName, lastName } = userEditForm.value;
+    const { email, password, firstName, lastName, organizatorName, phone} = userEditForm.value;
 
     const region = this.region;
   }
-  constructor(private authService: AuthService) {}
+  constructor(private authService: AuthService, private router: Router) {}
 
   ngOnInit() {
     if(this.authService.getUser()){
```
Comment (Getsov) [inline src/app/user/profile/profile.page.ts]: I still don’t have an answer - why calling the function twice if true?

File: src/app/user/profile/profile.page.ts (hunk 12 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.ts:17]: In the update function, I will check what the user is and submit only what I need.

File: src/app/user/profile/profile.page.html (hunk 13 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.html]: done

File: src/app/user/profile/profile.page.ts (hunk 14 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.ts]: done

File: src/app/user/profile/profile.page.ts (hunk 15 above)
Comment (Nikolay-Gavazov) [reply src/app/user/profile/profile.page.ts]: It was late, now I did it better.

**Review by Getsov — APPROVED** [review]

**Pull request merged**